lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -287,3 +304,34 @@ follow nl tr cps yield rollback header =
                         "Failed to roll backward: " <> show e
                 Right () -> do
                     step cursor'
+
+onSameTip :: BlockHeader -> BlockHeader -> Bool
+onSameTip = (==)
+
+data ErrInconsistentTips
+    = ErrInconsistentTips
+    | ErrTipNotFound ErrNetworkTip
+
+atomically
+    :: forall m a t block e. Monad m
+    => (BlockHeader -> BlockHeader -> Bool)
+    -- ^ Compare the preceding, and succeding tip, and judge whether the result
+    -- of the action can be trusted.
```

Comment:
I'd actually remove that parameter. There's no need to provide any particular predicate other then `==`, so we can cut off some indirection / complexity here :+1: